File: VB/CustomActionControl.Module/Module.vb
Imports System
Imports System.Collections.Generic
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Data.Filtering

Namespace CustomActionControl.Module
    Public NotInheritable Partial Class CustomActionControlModule
        Inherits ModuleBase

        Public Sub New()
            InitializeComponent()
            BaseObject.OidInitializationMode = OidInitializationMode.AfterConstruction
        End Sub
        Public Overrides Function GetModuleUpdaters(ByVal objectSpace As IObjectSpace, ByVal versionFromDB As Version) As IEnumerable(Of ModuleUpdater)
            Dim updater As ModuleUpdater = New Updater(objectSpace, versionFromDB)
            Return New ModuleUpdater() { updater }
        End Function
    End Class
    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            CreateOrder("0001", True, Date.Now)
            CreateOrder("0002", True, Date.Now.AddDays(-1))
            CreateOrder("0003", False, Date.Now.AddDays(-2))
            CreateOrder("0004", False, Date.Now)
            CreateOrder("0005", True, Date.Now.AddDays(-3))
        End Sub
        Private Sub CreateOrder(ByVal orderNo As String, ByVal isActive As Boolean, ByVal [date] As Date)
            Dim order As Order = ObjectSpace.FindObject(Of Order)(New BinaryOperator("OrderNo", orderNo))
            If order Is Nothing Then
                order = ObjectSpace.CreateObject(Of Order)()
                order.OrderNo = orderNo
                order.Active = isActive
                order.Date = [date]
            End If
        End Sub
    End Class
End Namespace

File: CS/CustomActionControl.Module/Module.cs
using System;
using System.Collections.Generic;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Updating;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Data.Filtering;

namespace CustomActionControl.Module {
    public sealed partial class CustomActionControlModule : ModuleBase {
        public CustomActionControlModule() {
            InitializeComponent();
            BaseObject.OidInitializationMode = OidInitializationMode.AfterConstruction;
        }
        public override IEnumerable<ModuleUpdater> GetModuleUpdaters(IObjectSpace objectSpace, Version versionFromDB) {
            ModuleUpdater updater = new Updater(objectSpace, versionFromDB);
            return new ModuleUpdater[] { updater };
        }
    }
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) : base(objectSpace, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            CreateOrder("0001", true, DateTime.Now);
            CreateOrder("0002", true, DateTime.Now.AddDays(-1));
            CreateOrder("0003", false, DateTime.Now.AddDays(-2));
            CreateOrder("0004", false, DateTime.Now);
            CreateOrder("0005", true, DateTime.Now.AddDays(-3));
        }
        private void CreateOrder(string orderNo, bool isActive, DateTime date) {
            Order order = ObjectSpace.FindObject<Order>(new BinaryOperator("OrderNo", orderNo));
            if (order == null) {
                order = ObjectSpace.CreateObject<Order>();
                order.OrderNo = orderNo;
                order.Active = isActive;
                order.Date = date;
            }
        }
    }
}
